---
title: "GC Bias Analysis by Group"
author: "ML Mansego"
output:
  pdf_document:
    toc: true
    toc_depth: '2'
    latex_engine: xelatex
  html_document:
    toc: true
    toc_depth: 2
    number_sections: true
    code_folding: hide
    theme: readable
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = "/mnt/mydisk/EM_Seq_ALSvsCT/fragment_analysis/")
sink("session_info_GC.txt")
sessionInfo()
sink()
library(tidyverse)
library(knitr)
library(readr)
library(dplyr)
library(tidyr)
library(ggpubr)
library(patchwork)
library(cowplot)
```

# Introduction

To evaluate potential sequencing biases related to guanine-cytosine (GC) content, normalized coverage across GC bins was assessed using Picard’s *CollectGcBiasMetrics* on deduplicated BAM files. Profiles were aggregated by group (ALS vs. Control, and ALS phenotypes) and examined for systematic deviations. The GC bias distributions showed the expected unimodal patterns, with peak coverage centered around 40–50% GC content in both groups. No group-specific enrichment or depletion was observed, and summary statistics confirmed comparable library quality across samples. This assessment ensures that GC bias is unlikely to confound downstream comparisons.


## Load Metadata

The following table summarizes key metadata for each sample, including diagnostic group (ALS vs Control), gender, age, and ALS clinical phenotype (Table 1).

```{r, warning=FALSE, message=FALSE, echo=FALSE}
sample_metadata <- read_csv("../pData/pData_ALSvsCT.csv") %>%
  select(Sample_ID, Condition, Gender, Age, ALS_Phenotype) 

kable(sample_metadata, digits = 4, caption = "Sample metadata including condition, gender, age, and ALS phenotype.")

```
Table 1. Metadata used for group assignments in the GC bias and fragmentomic analyses. The dataset includes ALS and control samples annotated with sex, age, and clinical phenotype (bulbar or spinal onset).

## GC Bias Metrics

Normalized GC coverage metrics were extracted from individual `*_gc_bias_metrics.txt` files generated by Picard for each sample. A custom parsing function was used to locate the data table within each file and extract GC bin–specific metrics, including normalized coverage and associated metadata. The results were merged with the corresponding sample annotations (condition, phenotype, age, gender) to enable group-wise comparisons.

### Read and combine GC bias data

```{r, warning=FALSE, message=FALSE, echo=FALSE}

gc_files <- list.files("stats", pattern = "_gc_bias_metrics.txt$", full.names = TRUE)
read_gc <- function(file) {
  lines <- readLines(file)
  skip <- grep("^ACCUMULATION_LEVEL", lines) - 1
  df <- read_tsv(file, skip = skip, col_types = cols())
  df$sample <- gsub("_gc_bias_metrics.txt", "", basename(file))
  return(df)
}

gc_data <- map_dfr(gc_files, read_gc)
gc_data <- gc_data %>%
  left_join(sample_metadata, by = c("sample" = "Sample_ID"))
head(gc_data)
```

### Plot mean GC bias curve by group

The plot below shows the average normalized coverage across GC bins for ALS and control groups. As expected, both groups displayed a unimodal distribution centered around the moderate GC content range (35–75%), reflecting typical sequencing performance. No substantial differences were observed between conditions in terms of coverage amplitude or peak position. These results suggest comparable GC bias profiles across groups, indicating no evidence of systematic technical bias due to GC content.

```{r, warning=FALSE, message=FALSE, echo=FALSE}
gc_grouped <- gc_data %>%
  group_by(Condition, `GC`) %>%
  summarise(mean_norm_cov = mean(`NORMALIZED_COVERAGE`, na.rm = TRUE), .groups = "drop")

GC_by_group<-ggplot(gc_grouped, aes(x = `GC`, y = mean_norm_cov, color = Condition)) +
  geom_line(linewidth = 1) +
  labs(title = "GC Bias - Mean per Group",
       x = "GC%", y = "Normalized Coverage (mean)") +
  theme_minimal()

ggsave("figures/GC_by_group.png", GC_by_group, width = 12, height = 10, dpi = 300)

```

### Diferential GC%

```{r, warning=FALSE, message=FALSE, echo=FALSE}
gc_bin_stats <- gc_data %>%
  filter(`GC` <= 100) %>%  
  group_by(`GC`) %>%
  summarise(
    mean_ALS = mean(NORMALIZED_COVERAGE[Condition == "ALS"], na.rm = TRUE),
    mean_Control = mean(NORMALIZED_COVERAGE[Condition == "Control"], na.rm = TRUE),
    mean_diff = mean_ALS - mean_Control,
    p.value = wilcox.test(NORMALIZED_COVERAGE ~ Condition, 
                          data = cur_data(), exact = FALSE)$p.value,
    .groups = "drop"
  )

# Add color to significant pvalue
figA <- ggplot(gc_bin_stats, aes(x = `GC`, y = mean_diff)) +
  geom_col(aes(fill = p.value < 0.05), width = 0.8) +
  geom_hline(yintercept = 0, linetype = "dashed") +
  scale_fill_manual(values = c("gray70", "#1b9e77")) +
  labs(
    title = "Normalized Coverage Across GC Content Bins",
    subtitle = "ALS - Control ",
    x = "GC Content (%)",
    y = "Mean Normalized Coverage Difference"
  ) +
  theme(legend.position = "none")

figA
```

## Mean Base Quality Across GC Content

```{r, warning=FALSE, message=FALSE, echo=FALSE}
# Average base quality by GC bin and condition
baseq_by_gc <- gc_data %>%
  group_by(Condition, GC) %>%
  summarise(mean_bq = mean(MEAN_BASE_QUALITY, na.rm = TRUE), .groups = "drop")

# Plot
figB <-ggplot(baseq_by_gc, aes(x = GC, y = mean_bq, color = Condition)) +
  geom_line(size = 1) +
  labs(
    title = "Mean Base Quality by GC Content",
    x = "GC Content (%)",
    y = "Mean Base Quality"
  ) +
  theme_minimal()

figB

```


## Bulbar vs Spinal onset ALS

```{r, warning=FALSE, message=FALSE, echo=FALSE}
gc_grouped_ALS <- gc_data %>%
  filter(Condition !="Control") %>%
  group_by(ALS_Phenotype, `GC`) %>%
  summarise(mean_norm_cov = mean(`NORMALIZED_COVERAGE`, na.rm = TRUE), .groups = "drop")

figC<-ggplot(gc_grouped_ALS, aes(x = `GC`, y = mean_norm_cov, color = ALS_Phenotype)) +
  geom_line(size = 1) +
  labs(title = "GC Bias - Mean per Group",
       x = "GC%", y = "Normalized Coverage (mean)") +
  theme_minimal()

gc_bin_stats_ALS <- gc_data %>%
  filter(Condition !="Control") %>%
  filter(`GC` <= 100) %>%  
  group_by(`GC`) %>%
  summarise(
    mean_Bulbar = mean(NORMALIZED_COVERAGE[ALS_Phenotype == "Bulbar"], na.rm = TRUE),
    mean_Spiral = mean(NORMALIZED_COVERAGE[ALS_Phenotype == "Spinal"], na.rm = TRUE),
    mean_diff = mean_Bulbar - mean_Spiral,
    p.value = wilcox.test(NORMALIZED_COVERAGE ~ ALS_Phenotype, 
                          data = cur_data(), exact = FALSE)$p.value,
    .groups = "drop"
  )

# Add color to significant pvalue
figD <- ggplot(gc_bin_stats_ALS, aes(x = `GC`, y = mean_diff)) +
  geom_col(aes(fill = p.value < 0.05), width = 0.8) +
  geom_hline(yintercept = 0, linetype = "dashed") +
  scale_fill_manual(values = c("gray70", "#1b9e77")) +
  labs(
    title = "Normalized Coverage Across GC Content Bins",
    subtitle = "Bulbar - Spinal Onset ALS",
    x = "GC Content (%)",
    y = "Mean Normalized Coverage Difference"
  ) +
  theme(legend.position = "none")
figC+figD


```

## GC Bias Summary by Group


```{r, warning=FALSE, message=FALSE, echo=FALSE}
gc_summary <- gc_data %>%
  group_by(Condition) %>%
  summarise(
    mean_gc = mean(`GC`, na.rm = TRUE),
    median_gc = median(`GC`, na.rm = TRUE),
    mean_norm_cov = mean(`NORMALIZED_COVERAGE`, na.rm = TRUE),
    sd_norm_cov = sd(`NORMALIZED_COVERAGE`, na.rm = TRUE)
  ) 
kable(gc_summary, digits = 4, caption = "Summary of GC Content and Normalized Coverage by Condition")
```
Table 2. Summary statistics of GC content and normalized coverage stratified by condition. Mean and median GC content were comparable between ALS and control samples. Similarly, mean normalized coverage and its standard deviation did not show substantial deviations between groups, indicating no significant global bias.

# Discussion and conclusion

The assessment of GC bias confirmed that the EM-seq libraries from plasma cfDNA in both ALS patients and controls were of comparable quality in terms of coverage uniformity across GC content. Although minor variability was observed in high-GC regions within the ALS group, these deviations were not statistically significant and likely reflect biological or technical heterogeneity rather than systematic bias.

These findings support the validity of downstream methylation and fragmentation analyses by excluding GC bias as a major confounding factor in this dataset.


```{r, warning=FALSE, message=FALSE, echo=FALSE}
supp_fig <- plot_grid(
  figA, figB,
  figC, figD,
  labels = c("A", "B", "C", "D"),
  label_size = 16,
  label_fontface = "bold",
  label_x = 0.02,  
  label_y = 0.98,   
  hjust = 0,        
  vjust = 1,       
  ncol = 2
)
ggsave("figures/Supplementary_Figure_1.png", supp_fig, width = 12, height = 10, dpi = 300)
```